### PR TITLE
add support clothing equip/unequip doafters

### PR DESCRIPTION
**Pull request by EmoGarbage404**
## About the PR

not for upstream, just a nice balance stick that people talk about a lot

## Media

https://github.com/space-wizards/space-station-14/assets/98561806/09d73c02-04db-495b-952a-ef26bd8f741c

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

no cl no fun

### CODE REVIEW

File: Content.Shared/Inventory/InventorySystem.Equip.cs [hunk 1]
```diff
@@ -149,14 +151,43 @@ private void OnUseSlot(UseSlotNetworkMessage ev, EntitySessionEventArgs eventArg
...
+                BreakOnHandChange = true,
+                BreakOnUserMove = true,
+                BreakOnTargetMove = true,
+                CancelDuplicate = true,
+                RequireCanInteract = true,
+                NeedHand = true
+            };
+
+            if (_gameTiming.IsFirstTimePredicted)
+                _doAfter.TryStartDoAfter(args);
```
Comment (metalgearsloth) [inline Content.Shared/Inventory/InventorySystem.Equip.cs:179]: I thought starting do_afters in prediction was supported.

File: Content.Shared/Inventory/InventorySystem.Equip.cs [hunk 2]
```diff
@@ -149,14 +151,43 @@ private void OnUseSlot(UseSlotNetworkMessage ev, EntitySessionEventArgs eventArg
             return false;
         }
 
+        if (checkDoafter &&
+            clothing != null &&
+            clothing.EquipDelay > TimeSpan.Zero &&
+            clothing.Slots.HasFlag(slotDefinition.SlotFlags) &&
+            _containerSystem.CanInsert(itemUid, slotContainer))
+        {
```
Comment (metalgearsloth) [inline Content.Shared/Inventory/InventorySystem.Equip.cs:159]: HasFlag allocates and sucks, use bitmask check.

File: Content.Shared/Inventory/InventorySystem.Equip.cs [hunk 3]
```diff
@@ -364,6 +399,34 @@ public bool CanAccess(EntityUid actor, EntityUid target, EntityUid itemUid)
...
+                CancelDuplicate = true,
+                RequireCanInteract = true,
+                NeedHand = true
+            };
+
+            if (_gameTiming.IsFirstTimePredicted)
+                _doAfter.TryStartDoAfter(args);
+            return false;
+        }
+
```
Comment (metalgearsloth) [inline Content.Shared/Inventory/InventorySystem.Equip.cs:429]: As above.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by UbaserB** [comment]
i know you said not for upstream... but can you make it upstream?

**Comment by EmoGarbage404** [comment]
> i know you said not for upstream... but can you make it upstream?

someone is free to add it but uhh. i'm not fighting that fight today.

**Comment by Emisse** [comment]
bout to add hardsuit equip doafter and have someone physically show up at my house this time

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**